### PR TITLE
Upgrade improvements

### PR DESCRIPTION
**Pull request by vangheem**
- [x] upgrade castle through add-on interface
- [x] script to upgrade all sites on instance
- [x] integration tests between versions of castle
- [x] integration test data
- [x] fix organization of upgrade steps so plone detects upgrades properly

### CODE REVIEW

File: castle/cms/_scripts/create-site.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+try:
+    from castle.cms.browser.site.addsite import AddCastleSite
+except ImportError:
+    # required! This could be getting called from upgrade
+    # integration tests
+    from castle.cms.browser.addsite import AddCastleSite
+
+
+app = app  # noqa
```
Comment (obct537) [inline castle/cms/_scripts/create-site.py]: ?

Comment (vangheem) [reply castle/cms/_scripts/create-site.py]: app is magic global variable inserted into script. This is to trick linter to not complain about future refs to global in code.

File: castle/cms/_scripts/create-site.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    description='...')
+parser.add_argument('--site-id', dest='site_id', default='Castle')
+parser.add_argument('--delete', dest='delete', default=False)
+args, _ = parser.parse_known_args()
+
+user = app.acl_users.getUser('admin')
+newSecurityManager(None, user.__of__(app.acl_users))
+
+if args.delete and args.site_id in app.objectIds():
+    app.manage_delObjects([args.site_id])
```
Comment (obct537) [inline castle/cms/_scripts/create-site.py]: Feel like some feedback here would be useful....just in case

Comment (vangheem) [reply castle/cms/_scripts/create-site.py]: done.

**Pull request merged**